src/GLua/AG/PrettyPrint.ag

```diff
@@ -46,6 +46,8 @@ data PrettyPrintConfig = PPConfig {
     spaceAfterComma :: Bool,
     semicolons :: Bool,
     cStyle :: Bool,
+    removeExtraParens :: Bool,
+    removeAllExtraParens :: Bool,
```

Comment:
The difference between "Remove extra parens" and "remove _all_ extra parens" is confusing on the surface. It sounds like the former doesn't work properly and second one does. 

I think we can tie the first one to the warning, which warns about redundant parentheses. The pretty print option could be "Remove redundant parentheses". 

The more aggressive version could be "minimize parentheses" or something, not sure.

Reply:
Yeah I like those setting names. It could also be a single int setting, 0/1/2